src/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java

```diff
@@ -880,8 +880,10 @@ public boolean _execute() throws Exception {
       }
     }
     // SP-307, BISERVER-9688
-
     final MasterReport report = getReport();
+    // Set unlimited queryLimit for reports running in "Run in background mode" in PUC.
+    report.setQueryLimit( -1 );
```

Comment:
You wrote in the letter that "•	A report that is run via the “Run in background/Schedule” option in PUC’s file browser never applies the row-limit". This code addresses this case. If we will not reset the limit, then the report will be generated with limit that user set in RD.